### PR TITLE
zh-CN: update Glossary/JavaScript

### PR DESCRIPTION
**Pull request by PassionPenguin**

### CODE REVIEW

File: files/zh-cn/glossary/javascript/index.md [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
+  sourceCommit: 409978ee3d2edd2b045adfbe39769b2dae6f31ea
 ---
 
 {{GlossarySidebar}}
 
-JavaScript (JS) 是一种编程语言，为通常用于客户端（client-side）的网页动态脚本，不过，也常通过像[Node.js](https://nodejs.org/)这样的包，用于服务器端（{{Glossary("Server","server")}}-side）。
+JavaScript（JS）是一种程序设计语言，通常用于客户端（client-side）的网页动态脚本，不过，也常通过像 [Node.js](https://nodejs.org/) 这样的包，用于{{Glossary("Server","服务器")}}-端。
 
-不应该把 JavaScript 和 [Java](/zh-CN/docs/Web/JavaScript/Guide/Introduction#JavaScript_and_Java) 混淆。“Java”和“JavaScript”都是 Oracle 公司在美国和其他国家注册的商标，但是这两种编程语言在语法、语义和使用方面都明显不同。
+**不应该**把 JavaScript 和 [Java](https://zh.wikipedia.org/wiki/Java) 混淆。_“Java”_ 和 _“JavaScript”_ 都是 Oracle 公司在美国和其他国家注册的商标，但是这两种程序设计语言在语法、语义和使用方面都明显不同。
```
Comment (yin1999) [inline files/zh-cn/glossary/javascript/index.md:12]:
```suggestion
**不应该**把 JavaScript 和 [Java](https://zh.wikipedia.org/wiki/Java) 混淆。“_Java_”和“_JavaScript_”都是 Oracle 公司在美国和其他国家注册的商标，但是这两种程序设计语言在语法、语义和使用方面都明显不同。
```

File: files/zh-cn/glossary/javascript/index.md [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
 
-Brendan Eich（彼时受雇于 Netscape）为服务器端构想的语言 JavaScript，不久便在 1995 年 9 月被加入 Netscape Navigator 2.0。JavaScript 很快获得了成功，而 {{glossary("Microsoft Internet Explorer", "Internet Explorer 3.0")}} 也在 1996 年 8 月，引入了对 JavaScript 的支持，冠以 JScript 之名。
+JavaScript 主要用于浏览器中，让开发者可以通过{{Glossary("DOM", "文档对象模型")}}来操纵网页内容、通过 {{domxref("fetch()")}} API 从服务器获取内容、通过 {{Glossary("IndexedDB")}} 存储复杂数据、使用 {{Glossary("canvas")}} 绘制图形、通过各种 {{Glossary("API")}} 与设备交互等等。JavaScript 是世界上最广泛使用的语言之一，归功于浏览器中 {{Glossary("API")}} 的发展和性能提升。
 
-1996 年 11 月，Netscape 开始与 ECMA 国际化组织合作以使 JavaScript 成为行业标准。从此以后，标准化的 JavaScript 就被称为 ECMAScript 并由 ECMA-262 规范，其最新版（第十四版，ECMA2023）于 2023 年 6 月生效。
+## 起源与历史
+
+Brendan Eich（彼时受雇于 Netscape）为服务器端构想的语言 JavaScript，很快在 1995 年 9 月被加入 Netscape Navigator 2.0。JavaScript 很快获得了成功，而 {{glossary("Microsoft Internet Explorer", "Internet Explorer 3.0")}} 也在 1996 年 8 月，引入了对 JavaScript 的支持，冠以 JScript 之名。
```
Comment (yin1999) [inline files/zh-cn/glossary/javascript/index.md:18]:
```suggestion
Brendan Eich（彼时受雇于 Netscape）为服务器端构想的语言——JavaScript——很快在 1995 年 9 月被加入到 Netscape Navigator 2.0 中。JavaScript 很快获得了成功，而 {{glossary("Microsoft Internet Explorer", "Internet Explorer 3.0")}} 也在 1996 年 8 月，引入了对 JavaScript 的支持，冠以 JScript 之名。
```

File: files/zh-cn/glossary/javascript/index.md [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
 
-JavaScript 通常用于浏览器，使开发者能通过{{Glossary("DOM")}}来操纵网页内容、或透过{{Glossary("AJAX")}}与{{Glossary("IndexedDB")}}来操作数据；还可以用它在{{Glossary("canvas")}}上面绘图、通过各种{{Glossary("API","APIs")}}与运行浏览器的各种设备交互……等等。由于近年来的发展、以及各浏览器的{{Glossary("API","APIs")}}性能改善，JavaScript 成了世界上最常用的编程语言之一。
+1996 年 11 月，Netscape 开始与 ECMA 国际化组织合作以使 JavaScript 成为行业标准。从此以后，标准化的 JavaScript 就被称为 ECMAScript 并由 ECMA-262 规范，其最新版（第十四版，ECMA2023）于 2023 年 6 月生效。
 
-最近，JavaScript 的流行程度，随着除浏览器外最流行的跨平台 JavaScript 运行环境——[Node.js](https://nodejs.org/) 平台的成功而大大提升。Node.js 使开发者可以在 PC 上使用 JavaScript 作为脚本语言以自动化处理和构建功能完备的 {{Glossary("HTTP")}} 和 {{Glossary("Web Sockets")}} 服务器。
+最近，JavaScript 的流行程度，随着除浏览器外最流行的跨平台 JavaScript 运行环境——[Node.js](https://nodejs.org/) 平台的成功而大大提升。Node.js 使用 [Chrome 的 V8 JavaScript 引擎](<https://zh.wikipedia.org/wiki/V8_(JavaScript引擎)>)——使开发者可以在 PC 上使用 JavaScript 作为脚本语言以自动化处理和构建功能完备的 {{Glossary("HTTP")}} 和 {{Glossary("Web Sockets")}} 服务器。
```
Comment (yin1999) [inline files/zh-cn/glossary/javascript/index.md:22]:
```suggestion
最近，JavaScript 的流行程度，随着除浏览器外最流行的跨平台 JavaScript 运行环境——[Node.js](https://nodejs.org/) 平台的成功而大大提升。Node.js——使用 [Chrome 的 V8 JavaScript 引擎](<https://zh.wikipedia.org/wiki/V8_(JavaScript引擎)>)——使开发者可以在 PC 上使用 JavaScript 作为脚本语言以自动化处理和构建功能完备的 {{Glossary("HTTP")}} 和 {{Glossary("Web Sockets", " WebSocket")}} 服务器。
```

File: files/zh-cn/glossary/javascript/index.md [hunk 4]
```diff
@@ -1,27 +1,31 @@
...
 
 ## 参见
```
Comment (yin1999) [inline files/zh-cn/glossary/javascript/index.md:24]: 参见部分需要更新一下

Comment (PassionPenguin) [reply files/zh-cn/glossary/javascript/index.md:24]: 😧能指出一下吗，（我只发现codecademy）

- [JavaScript](https://en.wikipedia.org/wiki/JavaScript) on Wikipedia
- The [JavaScript Guide](/en-US/docs/Web/JavaScript/Guide) on MDN
- [The "javascripting" workshop on NodeSchool](https://nodeschool.io/#workshoppers)
- [The JavaScript course on codecademy.com](https://www.codecademy.com/catalog/language/javascript)
- [The latest ECMAScript standard](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/)
- The [JavaScript reference](/en-US/docs/Web/JavaScript/Reference) on MDN
- [The _Eloquent JavaScript_ book](https://eloquentjavascript.net/)

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**